### ISSUE TITLE
TOML parse error: Invalid table header: Duplicate key `version`

### ISSUE THREAD

**Issue by hugovk** [charliermarsh/ruff#1894]
Run `ruff .` with this `pyproject.toml` in an empty directory:

```toml
# pyproject.toml

[tool.hatch]
version.source = "vcs"

[tool.hatch.version.raw-options]
local_scheme = "no-local-version"
```

Outputs error only without `--isolated` in newest 0.0.222:

```console
$ ruff --version
ruff 0.0.222
$  ruff .
error TOML parse error at line 6, column 1
  |
6 | [tool.hatch.version.raw-options]
  | ^
Invalid table header
Duplicate key `version`

$ ruff . --isolated
warning: No Python files found under the given path(s)
```

This file is parsed fine by other tools, such as Hatchling, python-build, Black, isort etc. Fuller version at https://github.com/hugovk/norwegianblue/

Error both with and without `--isolated` in 0.0.219:

```console
$ ruff --version
ruff 0.0.219
$ ruff .
error TOML parse error at line 6, column 1
  |
6 | [tool.hatch.version.raw-options]
  | ^
Invalid table header
Duplicate key `version`

$ ruff . --isolated
error TOML parse error at line 6, column 1
  |
6 | [tool.hatch.version.raw-options]
  | ^
Invalid table header
Duplicate key `version`
```

Error only without `--isolated` in 0.0.220:

```console
$ ruff --version
ruff 0.0.220
$ ruff .
error TOML parse error at line 6, column 1
  |
6 | [tool.hatch.version.raw-options]
  | ^
Invalid table header
Duplicate key `version`

$ ruff . --isolated
warning: No Python files found under the given path(s)
```

**Comment by messense** [comment]
Looks like both `toml` and `toml_edit` fail to parse the toml content, see https://www.rustexplorer.com/b/vthrz3

```bash
$ ./playground
Err(
    Error {
        inner: ErrorInner {
            kind: Custom,
            line: Some(
                5,
            ),
            col: 0,
            at: Some(
                55,
            ),
            message: "duplicate key: `version`",
            key: [
                "tool",
                "hatch",
            ],
        },
    },
)
Err(
    TomlError {
        message: "TOML parse error at line 6, column 1\n  |\n6 | [tool.hatch.version.raw-options]\n  | ^\nInvalid table header\nDuplicate key `version`\n",
        line_col: Some(
            (
                5,
                0,
            ),
        ),
    },
)
```

**Comment by henryiii** [comment]
I'm not familiar with the Rust release process, will this be closed when https://github.com/toml-rs/toml/issues/439 is available in Ruff, so I can just watch this issue? :)

**Comment by charliermarsh** [comment]
Yeah it should be! I'll try upgrading. It looks like they just cut a release.

**Comment by charliermarsh** [comment]
Looks like #2040 should close this out.

**Comment by charliermarsh** [comment]
Sorry, a little confusion with respect to the `toml` crate -- the upgrade I performed _did_ solve this problem, but it had the potential to introduce some other incompatibilities with TOML 1. So I want to revert for now, just to minimize churn. I expect this to be closed out soon-ish, but we do need a new release. (See: #2058.)

**Comment by hugovk** [comment]
Thank you for the fix, I can confirm this is fixed in 0.0.231.

I'm impressed it only took 9 days for: report to Ruff, report upstream to Rust TOML library, report upstream to the language-agnostic TOML test suite, then the test suite updated and released, then the TOML library updated and released, and then Ruff updated and released!
